docs/src/sections/webapp_solution.adoc

```diff
@@ -0,0 +1,529 @@
+do serwera. Jeśli odświeżanie tokena się nie powiedzie, użytkownik jest wylogowany, a zarazem przekierowany
+przez _UserLayout.tsx_ lub _AdminLayout.tsx_ do strony logowania.
+Jeśli odświeżanie się powiedzie, kolejne jego uruchomienie zostaje zakolejkowane do wykonania za 10 minut.
+Ponieważ ważność tokena na serwerze jest ustalona na 15 minut, token w przeglądarce jest zawsze ważny.
+Udostępnia również funkcje pomocnicze, takie jak _login_, _logout_, w celu wyeliminowania powtórzeń.
+
+.Funkcje _login_ i _logout_ pliku UserContext.tsx.
+[source, tsx]
+----
+include::../../../webapp/src/components/Context/UserContext.tsx[lines=47..59]
+----
+
+.Odświeżanie tokena w pliku UserContext.tsx.
+[source, tsx]
+----
+include::../../../webapp/src/components/Context/UserContext.tsx[lines=61..79]
+----
+
+Context _UserContext.tsx_ obejmuje wszystkie widoki,
+tak żeby dane o użytkowniku nie zostały usunięte przy zmianie widoków.
+Do tego korzysta z pliku pomocniczego _src/utils/persist.ts_,
+aby dane o użytkowniku zostały zapisane nawet przy zamknięciu przeglądarki.
+
+Plik _persist.ts_ zawiera funkcje dotyczące _localStorage_,
+czyli API do długotrwałego zapisu danych dostępnego w przeglądarce.
+API _localStorage_ pozwala zapisać dane typu klucz wartość, ale klucz i wartość muszą być wartością tekstową.
+Zostały napisane funkcje, aby obejść to ograniczenie, zamieniając obiekty na wartość tekstową przy zapisywaniu
+i odwrotnie przy wczytywaniu.
+
+Funkcja _setLocalJson_ zamienia przyjętą wartość w ciąg tekstowy i zapisuje go w _localStorage_.
+
+.Funkcja setLocalJson w pliku persist.ts.
+[source, typescript]
+----
+include::../../../webapp/src/utils/persist.ts[lines=13..15]
+----
+
+Funkcja _getLocalJson_ pobiera wartość z _localStorage_ i zamienia ją z ciągu tekstowego na obiekt _JSON_,
+przy okazji zamieniając ciągi tekstowe na obiekty _Date_.
+
+.Funkcja getLocalJson pliku persist.ts.
+[source, typescript]
+----
+include::../../../webapp/src/utils/persist.ts[lines=4..11]
+----
+
+Ponieważ w Reakcie zmienne wartości muszą być przechowywane w stanie, został napisany hook _useLocalStorage_,
+który oprócz zapisywania wartości w stanie, nadpisuje ją w _localStorage_.
+Używa się go jak standardowego hooku _useState_, z tą różnicą, że zamiast przyjmować wartość, przyjmuje klucz,
+który wskazuje na wartość w _localStorage_.
+
+.Funkcja _useLocalStorage_ w pliku _persist.ts_.
+[source, typescript]
+----
+include::../../../webapp/src/utils/persist.ts[lines=17..30]
+----
+
+Pasek nawigacyjny po lewej stronie ekranu został zaimplementowany w _src/components/common/Sidebar/Sidebar.tsx_.
+Jest on dodawany do komponentów w _PanelLayout.tsx_.
+Każdy przycisk na tym pasku jest zdefiniowany przy pomocy komponentu _SidebarEntry.tsx_.
+Jest możliwość przekazania do niego argumentu _adminOnly_, jeśli przycisk ma być wyświetlony tylko dla adminów.
+Przycisk jest podświetlony, jeśli otwarta ścieżka w przeglądarce zawiera w sobie ścieżkę danego przycisku.
```

Comment:
zamiast powtarzać przycisk można dać "jego ścieżkę" czy coś, żeby to trochę zredagować i nie powtarzać